.github/get_results.php

```diff
@@ -1,68 +1,90 @@
-        'end' => $result['date_end']->format('Y-m-d H:i:s'),
-        'duration' => $result['duration'],
+        'start' => $globalResults['date_start']->format('Y-m-d H:i:s'),
+        'end' => $globalResults['date_end']->format('Y-m-d H:i:s'),
+        'duration' => $globalResults['duration'],
         'skipped' => 0,
         'pending' => 0,
-        'passes' => $testPassed ? 1 : 0,
-        'failures' => $testPassed ? 0 : 1,
+        'passes' => $globalResults['passes'],
+        'failures' => $globalResults['failures'],
         'suites' => 1,
-        'tests' => 1,
+        'tests' => count($results),
     ],
     'suites' => [
         'uuid' => uniqid(),
-        'title' => $result['title'],
+        'title' => $globalResults['title'],
         'file' => '',
-        'duration' => $result['duration'],
+        'duration' => $globalResults['duration'],
         'hasSkipped' => false,
         'hasPending' => false,
-        'hasPasses' => $testPassed > 0,
-        'hasFailures' => $testPassed === 0,
+        'hasPasses' => $globalResults['passes'] > 0,
+        'hasFailures' => $globalResults['failures'] > 0,
         'totalSkipped' => 0,
         'totalPending' => 0,
-        'totalPasses' => $testPassed ? 1 : 0,
-        'totalFailures' => $testPassed ? 0 : 1,
+        'totalPasses' => $globalResults['passes'],
+        'totalFailures' => $globalResults['failures'],
         'hasSuites' => true,
         'hasTests' => false,
         'tests' => [],
         'suites' => [[
             'uuid' => uniqid(),
-            'title' => $result['title'],
+            'title' => $globalResults['title'],
             'file' => '',
-            'duration' => $result['duration'],
+            'duration' => $globalResults['duration'],
             'hasSkipped' => false,
             'hasPending' => false,
-            'hasPasses' => $testPassed > 0,
-            'hasFailures' => $testPassed === 0,
+            'hasPasses' => $globalResults['passes'] > 0,
+            'hasFailures' => $globalResults['failures'] > 0,
             'totalSkipped' => 0,
             'totalPending' => 0,
-            'totalPasses' => $testPassed ? 1 : 0,
-            'totalFailures' => $testPassed ? 0 : 1,
+            'totalPasses' => $globalResults['passes'],
+            'totalFailures' => $globalResults['failures'],
             'hasSuites' => false,
             'hasTests' => true,
             'suites' => [],
-            'tests' => [$result],
+            'tests' => $results,
         ]],
     ],
 ];
 
 $filename = 'autoupgrade_' . date('Y-m-d') . '-' . $branch . '.json';
 file_put_contents($filename, json_encode($data));
 
-function getTestResultFromFile($file)
+function getResultFiles(string $branch): array
+{
+    $rii = new RecursiveIteratorIterator(new RecursiveDirectoryIterator('./artifacts'));
+    $files = [];
+    foreach ($rii as $file) {
+        if ($file->isDir() || strpos($file->getPathname(), $branch . '.txt') === false) {
```

Comment:
```suggestion
        if ($file->isDir() || strpos($file->getPathname(), $branch . '.txt') === false) {
```
IF you're looking for a file name ended by `$branch . '.txt'` it's maybe easier to use a regex to resolve the issue

Reply:
I think it’s simpler like that because branch name can contain characters that must be escaped if using regex, not needed when using `strpos`

Reply:
Don't forget https://www.php.net/manual/fr/function.preg-quote.php :D